driver-core/src/main/com/mongodb/internal/connection/SingleServerCluster.java

```diff
@@ -94,7 +95,9 @@ public void serverDescriptionChanged(final ServerDescriptionChangedEvent event)
                     }
                 }
             }
-            publishDescription(descriptionToPublish);
+            if (shouldPublishChangeEvent(event.getNewDescription(), event.getPreviousDescription())) {
```

Comment:
This isn't correct.  We still need to do everything in `publishDescription` _except fire the change event.

Reply:
I have moved the check to `publishDescription`.